### PR TITLE
Fix circuit size in `gen_proof`

### LINKED ISSUE #149: Wrong circuit size in `gen_proof`
`gen_proof` is updating the size of the `PI` struct with the result from `padded_circuit_size()`. The result of this call should just be used as an upper bound for the circuit to trim the SRS.
The actual size of the circuit is obtained via `prover.circuit_bound()`.

### PR DESCRIPTION
**Pull request by davidnevadoc**
Close #149

### CODE REVIEW

**Comment by davidnevadoc** [comment]
Maybe we should change the name `padded_circuit_size()`. It doesn't return the real size of the circuit (padded or unpadded), it just gives an upper bound to what its size can be. It is really easy to misuse.

File: plonk-core/src/circuit.rs [hunk 1]
```diff
@@ -280,7 +280,7 @@ where
         let mut prover = Prover::new(transcript_init);
         // Fill witnesses for Prover
         self.gadget(prover.mut_cs())?;
-        prover.cs.public_inputs.update_size(circuit_size);
+        prover.cs.public_inputs.update_size(prover.circuit_bound());
```
Comment (CPerezz) [inline plonk-core/src/circuit.rs:283]: Why not adding a fn for Prover like: 
`Prover::update_pi_size` for which the prover has the `circuit_bound` already. And we don't need to pass any parameters.

**Review by CPerezz — CHANGES_REQUESTED** [review]

File: plonk-core/src/circuit.rs (hunk 1 above)
Comment (davidnevadoc) [reply plonk-core/src/circuit.rs:283]: While trying this out I realized we may as well try to get rid of `n` in the `PI`. I've opened a separate PR for this #152 .

**Comment by davidnevadoc** [comment]
Fixed in #153 .

**Pull request closed without merging**